external/warc/src/main/java/com/digitalpebble/stormcrawler/warc/WARCRecordFormat.java

```diff
@@ -149,9 +268,13 @@ public static String getDigestSha1(byte[] bytes1, byte[] bytes2) {
         buffer.append("Content-Length").append(": ")
                 .append(Integer.toString(contentLength)).append(CRLF);
 
-        // TODO get actual fetch time from metadata if any
-        Date now = new Date();
-        buffer.append("WARC-Date").append(": ").append(WARC_DF.format(now))
+        // get actual fetch time from metadata if any
+        String captureTime = metadata.getFirstValue("_request.time_");
+        if (captureTime == null) {
+            Date now = new Date();
```

Comment:
use Instant instead of Date? Would mean changing in the rest of this class

Reply:
Good idea. It would actually only change these lines.